docs/dev/framework/templates/twig.md

```diff
@@ -0,0 +1,535 @@
+use whenever you do not know the exact namespace beforehand. This namespace
+will be substituted with a specific namespace when the templates are compiled.
+In each situation we're choosing the **next available** template that has a 
+**lower priority** than the current one.
+
+And yes, you can totally use this to extend, embed or include templates. Have a
+look at the following example to get an idea.
+
+#### Template hierarchy example
+In this example, we're dealing with four manifestations of the same
+`card.html.twig` template: two in bundles, two more in the application.  
+
+{{< tabs groupId="template-hierarchy-example">}}
+{{% tab name="a) card-bundle" %}}
+The original template of the `card-bundle`:
+
+```twig
+{# /vendor/foo/card-bundle/contao/templates/card.html.twig #}
+
+{% import '@ContaoCore/Image/Studio/_macros.html.twig' as studio %}
+
+<section class="card">
+  {% block card %}
+    <header class="title">
+      {% block title %}{{ title }}{% endblock %}
+    </header>     
+    <main>
+      {% block content %}
+        {{ studio.figure(figure) }}
+        {{ description|raw }}
+      {% endblock %}
+    </main>
+    <footer>
+      {% block footer %}<p class="author">by {{ author }}</p>{% endblock %}
+    </footer
+  {% endblock %}
+</section>
+```
+{{% /tab %}}
+
+{{% tab name="b) card-time-bundle" %}} 
+A `card-time-bundle` extending the original bundle and adding information to
+the footer - this bundle was loaded *after* the `card-bundle`, therefore it is
+further up in our template hierarchy:
+
+```twig
+{# /vendor/bar/card-time-bundle/contao/templates/card.html.twig #}
+
+{% extends '@Contao/card' %}
+
+{% block footer %}
+  {{ parent() }}
+  <p class="last-modified">edited at {{ modified_at|ago }}</p>
+{% endblock %}
+```
+{{% /tab %}}
+
+{{% tab name="c) global template" %}}
+The `card` template of the global template folder adding some wrappers, 
+because, you know, you can't have enough *divs*.
+
+```twig
+{# /templates/card.html.twig #}
+
+{% extends '@Contao/card' %}
+
+{% block title %}<div class="inner">{{ parent() }}</div>{% endblock %}
+{% block card %}<div class="inner">{{ parent() }}</div>{% endblock %}
+```
+{{% /tab %}}
+
+{{% tab name="d) emoji theme" %}}
+And finally the application's `emoji` theme adding, well, …
+
+```twig
+{# /templates/emoji/card.html.twig #}
+   
+{% extends '@Contao/card' %}
+{% block title %}🤩 {{ parent() }} 🤯{% endblock %}
+```
+{{% /tab %}}
+{{< /tabs >}}
+
+Resolving all *extends* in the right order would effectively yield the
+following template - note how each stage can adjust/contribute to blocks
+without the need to know about the others because every *extend* uses the
+managed `@Contao` namespace:
+
+```twig
+{% import '@ContaoCore/Image/Studio/_macros.html.twig' as studio %}
+
+<section class="card">
+  <div class="inner">
+    <header class="title">
+      <div class="inner">🤩 {{ title }} 🤯</div>
```

Comment:
My brain-based parser outputs this the other way round.
Is it wrong in the documentation, the implementation or my brain? ☺️
```suggestion
      🤩 <div class="inner">{{ title }}</div> 🤯
```

Reply:
Eh, of course :smile:.  Must have been me using brain-based parsing when writing.